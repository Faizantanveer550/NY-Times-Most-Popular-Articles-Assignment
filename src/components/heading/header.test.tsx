import "@testing-library/jest-dom/extend-expect"; // Optional import for extended matchers
import { render, screen } from "@testing-library/react";
import Heading from "./heading";

test("Snapshot Test", () => {
  const title = "My Heading";

  const { asFragment } = render(<Heading title={title} />);
  expect(asFragment()).toMatchSnapshot();
});

test("Heading renders the title prop", () => {
  const title = "My Heading";
  render(<Heading title={title} />);

  expect(screen.getByText(title)).toBeInTheDocument();
});

test("Heading renders default title when no prop is provided", () => {
  render(<Heading />); // No title prop

  expect(screen.getByText("Unnamed Heading")).toBeInTheDocument(); // Adjust default text as needed
});

test("Heading applies the correct styles", () => {
  const title = "Styled Heading";
  render(<Heading title={title} />);

  // Assuming the styled component adds a class like 'heading'
  expect(screen.getByText(title)).toHaveClass("heading");
});
