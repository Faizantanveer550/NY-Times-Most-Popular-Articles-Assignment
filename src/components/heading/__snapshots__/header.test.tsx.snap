// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="sc-aYaIB kGpvrW heading"
  >
    My Heading
  </div>
</DocumentFragment>
`;
